rust/src/lib.rs

```diff
@@ -112,18 +113,24 @@ impl Transaction {
         self.witness_set.clone()
     }
 
+    pub fn is_valid(&self) -> bool {
+        self.is_valid.clone()
+    }
+
     pub fn auxiliary_data(&self) -> Option<AuxiliaryData> {
         self.auxiliary_data.clone()
     }
 
     pub fn new(
         body: &TransactionBody,
         witness_set: &TransactionWitnessSet,
+        is_valid: bool,
```

Comment:
I think we shouldn't have this as a parameter and instead if advances users for some reason need to have it as false we can instead add in a `pub fn set_is_valid(&mut self, is_valid: bool)`.